tools/android/packaging/xbmc/src/org/xbmc/kodi/Main.java.in

```diff
@@ -53,51 +54,27 @@ public class Main extends NativeActivity implements Choreographer.FrameCallback
   public Main()
   {
     super();
+    MainActivity = this;
   }
 
-  public Surface getVideoViewSurface()
-  {
-    return mVideoView.getSurface();
-  }
-
-  public Rect getVideoViewSurfaceRect()
+  public Rect getDisplayRect()
   {
     Rect ret = new Rect();
     ret.top = 0;
     ret.left = 0;
-    ret.right = mVideoView.mWidth;
-    ret.bottom = mVideoView.mHeight;
+    ret.right = 0;
+    ret.bottom = 0;
 
-    return ret;
-  }
-
-  public void setVideoViewSurfaceRect(final int left, final int top, final int right, final int bottom)
-  {
-    runOnUiThread(new Runnable()
+    try
     {
-      @Override
-      public void run()
-      {
-        android.widget.RelativeLayout.LayoutParams mp = new android.widget.RelativeLayout.LayoutParams(mVideoView.getLayoutParams());
-        mp.setMargins(left, top, mVideoLayout.getWidth() - right, mVideoLayout.getHeight() - bottom);
-        mVideoView.setLayoutParams(mp);
-        mVideoView.requestLayout();
-      }
-    });
-  }
+      ret.right = thisView.getRootView().getWidth();
+      ret.bottom = thisView.getRootView().getHeight();
+    }
+    catch (Exception e) {}
```

Comment:
Unlikely, indeed. I'm *forced* to do a try-catch, because "something" could throw "some exception" in the block, but there is actually 0 chance that it could fail, really